components/contexts/Web3Context.js

```diff
@@ -24,12 +19,13 @@ const providerOptions = {
 
 export const Web3Provider = ({ children }) => {
   const [web3Modal, setWeb3Modal] = useState()
-  const [accounts, setAccounts] = useState([])
+  const [signer, setSigner] = useState()
+  const [address, setAddress] = useState()
```

Comment:
Switching accounts array (with only one account anyways) to signer and address for separate use.